fix: macos is now platform specific

rm-kierancoker opened the pull request:
it looks like with the latest 0.10.0 release, neovim is publishing macos-arm or macos-x86_64 versions only

relates to issue #21

rm-kierancoker committed:
fix: macos is now platform specific

it looks like with the latest 0.10.0 release, neovim is publishing
macos-arm or macos-x86_64 versions only

tedkulp commented:
Wouldn't this need to check for versions below 0.10 and still use the old logic?

rm-kierancoker commented:
Possibly, I thought the plugin was version released but if we'd prefer that approach, sure happy to make that change =]

rm-kierancoker committed:
fix: add a check for version less that 0.10.x

If the version is less than 0.10.x we will want to use the old platform
of just `macos` rather than split based on arch

rm-kierancoker committed:
fix: use new arch on stable or nightly versions

rm-kierancoker commented:
I've implemented the check for version, testing upgrading to `0.10.0` (removing and old version) then going back to `0.9.0` - this has successfully downloaded and installed both `v0.9.0` and `v0.10.0`

tedkulp commented:
Working here. Thank you for that. Hopefully it gets merged in soon.

If anyoone can't wait: `asdf plugin update neovim pull/22/head` worked for me

richin13 commented on bin/utils.sh:
```diff
@@ -31,25 +31,30 @@ download_url() {
-      fi
-      ;;
-  esac
+	case "$(uname -s)" in
+	"Linux")
+		platform=linux64
+		;;
+	"Darwin")
+		# Check version is less than v0.10.0 - if it is, use the old platform name
+		# using IFS to split the version number into an array
+		IFS='.' read -r -a version_array <<<"${version#v}"
+
+		# version_array[0] is the major version and version_array[1] is the minor version
+		if ((version_array[0] == 0 && version_array[1] < 10)) && [[ ! $version =~ ^stable|nightly$ ]]; then
+			platform=macos
+		else
+			case "$(uname -p)" in
+			"arm")
+				platform=macos-arm64
+				;;
+			*)
+				platform=macos-x86_64
+				;;
+			esac
+		fi
+		;;
+	esac
```
Thank you very much! Could you please fix the indentation here? Looks like the case has an extra level 😅

richin13 replied:
Nevermind, this is blocking people. I'll update it later

richin13 commented:
Thank you @rm-kierancoker

rm-kierancoker commented:
Apologies for the formatting, not entirely sure how i didn't recognise it was so massively different 😅 

Thanks for merging and making this plugin ❤️
